ValidationLibrary/Rules/HasNewestPtcsJenkinsLibRule.cs

```diff
@@ -96,9 +133,40 @@ private async Task Fix(IGitHubClient client, Repository repository)
             var commitResponse = await client.Git.Commit.Create(repository.Owner.Login, repository.Name, commit);
 
             var refUpdate = new ReferenceUpdate(commitResponse.Sha);
-            await client.Git.Reference.Update(repository.Owner.Login, repository.Name, $"heads/{branchName}", refUpdate);
+            return await client.Git.Reference.Update(repository.Owner.Login, repository.Name, $"heads/{BranchName}", refUpdate);
+        }
 
-            var pullRequest = new NewPullRequest($"[Automatic Validation] Update {LibraryName} to latest version.", branchReference.Ref, master.Ref)
+        private async Task<string> GetFixedContent(IGitHubClient client, Repository repository, string branchName)
+        {
+            _logger.LogTrace("Rule {ruleClass} / {ruleName}: Retrieving fixed contents for JenkinsFile from branch {branch}", nameof(HasNewestPtcsJenkinsLibRule), RuleName, branchName);
+            var jenkinsContent = await GetJenkinsFileContent(client, repository, branchName);
+            if (jenkinsContent == null)
+            {
+                _logger.LogWarning("Rule {ruleClass} / {ruleName}, no {filename} found, unable to fix.",
+                    nameof(HasNewestPtcsJenkinsLibRule), RuleName, JenkinsFileName);
+                return null;
+            }
+            var fixedContent = _regex.Replace(jenkinsContent.Content, $"'{LibraryName}@{_expectedVersion}'");
+            return string.Equals(fixedContent, jenkinsContent.Content) ? null : fixedContent;
```

Comment:
What `StringComparison` options would best be suited for this comparison? Consider specifying it explicitly.